src/main/java/de/caritas/cob/userservice/api/tenant/TenantContext.java

```diff
@@ -9,18 +9,35 @@ private TenantContext() {
 
   }
 
-  private static final ThreadLocal<Long> CURRENT_TENANT = new ThreadLocal<>();
-
+  private static final ThreadLocal<TenantData> CURRENT_TENANT_DATA = new ThreadLocal<>();
 
   public static Long getCurrentTenant() {
-    return CURRENT_TENANT.get();
+    return CURRENT_TENANT_DATA.get() != null ? CURRENT_TENANT_DATA.get().getTenantId() : null;
+  }
+
+  public static TenantData getCurrentTenantData() {
+    return CURRENT_TENANT_DATA.get();
+  }
+
+  public static void setCurrentTenantData(TenantData tenantData) {
+    CURRENT_TENANT_DATA.set(tenantData);
+  }
+
+  public static void setCurrentTenant(Long tenantId) {
+    if (CURRENT_TENANT_DATA.get() == null) {
+      CURRENT_TENANT_DATA.set(new TenantData());
+    }
+    CURRENT_TENANT_DATA.get().setTenantId(tenantId);
   }
 
-  public static void setCurrentTenant(Long tenant) {
-    CURRENT_TENANT.set(tenant);
+  public static void setCurrentSubdomain(String subdomain) {
+    if (CURRENT_TENANT_DATA.get() == null) {
```

Comment:
this block could be extracted to a method initializeCurrentTenantDataIfNotExist()

Reply:
done